### ISSUE TITLE
🐛 Bug Report: Error 401 in log nginx swag proxy

### ISSUE THREAD

**Issue by babsz02** [stonith404/pingvin-share#401]
### 👟 Reproduction steps

Hello,
I'm having trouble generating 401 errors in my logs that are triggering my fail2ban and banning the ip's of people downloading files.

I just sent the link but I still get requests on /api/auth/token that return 401 errors and trigger the nginx-unauthorized jail.

here are the anonimysed logs : 

178.xx.32.xxx - - [06/Feb/2024:11:19:29 +0100] "POST /api/auth/token HTTP/2.0" 401 43 "https://xxxxxx.xxxxxxxx.fr/share/Y0NzIzN" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
178.xx.32.xxx - - [06/Feb/2024:11:19:59 +0100] "POST /api/auth/token HTTP/2.0" 401 43 "https://xxxxxx.xxxxxxxx.fr/share/Y0NzIzN" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
178.xx.32.xxx - - [06/Feb/2024:11:20:03 +0100] "POST /api/auth/token HTTP/2.0" 401 43 "https://xxxxxx.xxxxxxxx.fr/share/Y0NzIzN" "Mozilla/5.0 (iPhone; CPU iPhone OS 17_2_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148 [FBAN/FBIOS;FBAV/437.0.1.29.232;FBBV/543605699;FBDV/iPhone16,1;FBMD/iPhone;FBSN/iOS;FBSV/17.2.1;FBSS/3;FBCR/;FBID/phone;FBLC/fr;FBOP/80]"
178.xx.32.xxx - - [06/Feb/2024:11:20:29 +0100] "POST /api/auth/token HTTP/2.0" 401 43 "https://xxxxxx.xxxxxxxx.fr/share/Y0NzIzN" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
178.xx.32.xxx- - [06/Feb/2024:11:20:53 +0100] "GET /api/shares/Y0NzIzN/files/954e3d56-ae31-43ec-ae60-27875722b5a7 HTTP/2.0" 200 1092626495 "https://xxxxxx.xxxxxxxx.fr/share/Y0NzIzN" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
178.xx.32.xxx - - [06/Feb/2024:11:20:59 +0100] "POST /api/auth/token HTTP/2.0" 401 43 "https://xxxxxx.xxxxxxxx.fr/share/Y0NzIzN" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"

### 👍 Expected behavior

no 401 error creation

### 👎 Actual Behavior

ban ip fail2ban

### 🌐 Browser

chrome

**Comment by axiomen** [comment]
Hah, literally what I found out today in this issue #354. I wouldn't be surprised if these two are the same issues.

**Comment by babsz02** [comment]
Yes, I also just saw number @https://github.com/stonith404/pingvin-share/issues/354 and it relieved me to see that others had the same problem.

I always have problems of 401 error generate in cascade but I also had a doubt the buffer of Swag proxy.

I tried to set an ignoreregex in the nginx-unautorized filter configuration but it didn't work.

PS I deactivated my jail nginx-unautorized and it works with a VPN and not on my phone ne 5G :/


here is the pingvin log : 

[Nest] 50  - 02/06/2024, 12:59:38 PM   ERROR [ExpressAdapter] Premature close
Error [ERR_STREAM_PREMATURE_CLOSE]: Premature close
    at ServerResponse.onclose (node:internal/streams/end-of-stream:159:30)
    at ServerResponse.emit (node:events:526:35)
    at emitCloseNT (node:_http_server:1023:10)
    at Socket.onServerResponseClose (node:_http_server:278:5)
    at Socket.emit (node:events:526:35)
    at TCP.<anonymous> (node:net:337:12)


log to access.log Nginx Swag


84.17.43.244 - - [06/Feb/2024:14:04:54 +0100] "POST /api/auth/token HTTP/2.0" 401 43 "https://transfert.xxxxxxxxx.fr/share/Y0NzIzN" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
84.17.43.244 - - [06/Feb/2024:14:04:56 +0100] "GET /api/shares/Y0NzIzN HTTP/2.0" 304 0 "https://transfert.xxxxxxxxx.fr/share/Y0NzIzN" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
84.17.43.244 - - [06/Feb/2024:14:04:56 +0100] "GET /_next/data/nrqVFxuVPBxdreAOgrpgr/index.json HTTP/2.0" 200 2 "https://transfert.xxxxxxxxx.fr/share/Y0NzIzN" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
84.17.43.244 - - [06/Feb/2024:14:04:56 +0100] "GET /_next/data/nrqVFxuVPBxdreAOgrpgr/auth/signIn.json HTTP/2.0" 200 2 "https://transfert.xxxxxx.fr/share/Y0NzIzN" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
84.17.43.244 - - [06/Feb/2024:14:04:57 +0100] "GET /sw.js HTTP/2.0" 304 0 "https://transfert.xxxxxxxxx.fr/sw.js" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
84.17.43.244 - - [06/Feb/2024:14:04:58 +0100] "GET /workbox-77a77757.js HTTP/2.0" 304 0 "https://transfert.xxxxxxxxx.fr/sw.js" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
84.17.43.244 - - [06/Feb/2024:14:04:59 +0100] "GET /sw.js HTTP/2.0" 304 0 "https://transfert.xxxxxxxxx.fr/sw.js" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
84.17.43.244 - - [06/Feb/2024:14:04:59 +0100] "GET /workbox-77a77757.js HTTP/2.0" 304 0 "https://transfert.xxxxxxxxx.fr/sw.js" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
84.17.43.244 - - [06/Feb/2024:14:05:10 +0100] "POST /api/auth/token HTTP/2.0" 401 43 "https://transfert.xxxxxxxxx.fr/auth/signIn" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
84.17.43.244 - - [06/Feb/2024:14:05:26 +0100] "POST /api/auth/token HTTP/2.0" 401 43 "https://transfert.xxxxxxxxx.fr/share/Y0NzIzN" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
84.17.43.244 - - [06/Feb/2024:14:05:47 +0100] "GET /api/shares/Y0NzIzN/files/954e3d56-ae31-43ec-ae60-27875722b5a7 HTTP/2.0" 200 2484567711 "https://transfert.xxxxxxxxx.fr/share/Y0NzIzN" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
192.168.1.254 - - [06/Feb/2024:14:05:57 +0100] "POST /api/auth/token HTTP/2.0" 401 43 "https://transfert.xxxxxxxxx.fr/share/Y0NzIzN" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
84.17.43.244 - - [06/Feb/2024:14:06:10 +0100] "GET /api/websocket HTTP/1.1" 101 149878 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
84.17.43.244 - - [06/Feb/2024:14:06:10 +0100] "GET /api/websocket HTTP/1.1" 101 149856 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"


log error.log ngnix swag

2024/02/06 13:54:17 [error] 478#478: *48 upstream prematurely closed connection while reading upstream, client: 80.214.96.171, server: transfert.*, request: "GET /api/shares/Y0NzIzN/files/954e3d56-ae31-43ec-ae60-27875722b5a7 HTTP/2.0", upstream: "http://192.168.1.10:3002/api/shares/Y0NzIzN/files/954e3d56-ae31-43ec-ae60-27875722b5a7", host: "transfert.xxxxxxxx.fr", referrer: "https://transfert.xxxxxxxx.fr/share/Y0NzIzN"
2024/02/06 13:57:10 [crit] 478#478: connect() to [2a02:26f0:b80:3::210:9597]:80 failed (99: Address not available) while requesting certificate status, responder: r3.o.lencr.org, peer: [2a02:26f0:b80:3::210:9597]:80, certificate: "/config/keys/cert.crt"
2024/02/06 14:01:21 [error] 478#478: *14 upstream prematurely closed connection while reading upstream, client: 80.214.96.171, server: transfert.*, request: "GET /api/shares/Y0NzIzN/files/954e3d56-ae31-43ec-ae60-27875722b5a7 HTTP/2.0", upstream: "http://192.168.1.10:3002/api/shares/Y0NzIzN/files/954e3d56-ae31-43ec-ae60-27875722b5a7", host: "transfert.xxxxxxxx.fr", referrer: "https://transfert.xxxxxxxx.fr/share/Y0NzIzN"

**Comment by axiomen** [comment]
@babsz02 These are two separate, but related issues, I'd say.

The `Premature close` error from what I've found happens due to Nginx packaged with the pingvin-share docker image. Try the current development image ([stonith404/pingvin-share:development](https://hub.docker.com/layers/stonith404/pingvin-share/development/images/sha256-60aae8dd810d644ca1fecb02ccfdc243a7e806cb19f71323bae2ee5738123664?context=explore)) if you can, which comes with Caddy instead. If it works for you, please mention it in #354.

**Comment by babsz02** [comment]
@axiomen 

Thank you for your reply, I have this error that appears in the logs now:

ERR ts=1707228546.04678 logger=http.handlers.reverse_proxy msg=aborting with incomplete response upstream=localhost:8080 duration=0.01668699 request={"remote_ip":"172.24.0.1","remote_port":"45630","client_ip":"172.24.0.1","proto":"HTTP/1.1","method":"GET","host":"transfert.xxxxxx.fr","uri":"/api/shares/Y0NzIzN/files/954e3d56-ae31-43ec-ae60-27875722b5a7","headers":{"X-Forwarded-Server":["transfert.xxxxx.fr"],"Sec-Fetch-Site":["same-origin"],"Cookie":[],"X-Forwarded-Method":["GET"],"Referer":["https://transfert.xxxxxx.fr/share/Y0NzIzN"],"X-Forwarded-For":["172.24.0.1"],"X-Forwarded-Host":["transfert.xxxxx.fr"],"X-Original-Url":["https://transfert.xxxx.fr/api/shares/Y0NzIzN/files/954e3d56-ae31-43ec-ae60-27875722b5a7"],"X-Forwarded-Ssl":["on"],"X-Forwarded-Port":["443"],"Accept-Encoding":["gzip, deflate, br"],"Sec-Fetch-Mode":["navigate"],"X-Forwarded-Proto":["http"],"X-Original-Method":["GET"],"User-Agent":["Mozilla/5.0 (iPhone; CPU iPhone OS 17_4 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/121.0.6167.138 Mobile/15E148 Safari/604.1"],"Sec-Fetch-Dest":["document"],"Accept-Language":["fr-FR,fr;q=0.9"],"Accept":["text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"],"X-Forwarded-Uri":["/api/shares/Y0NzIzN/files/954e3d56-ae31-43ec-ae60-27875722b5a7"],"X-Real-Ip":["80.214.97.184"]}} error=writing: write tcp 172.24.0.2:3000->172.24.0.1:45630: write: broken pipe
Error: Request aborted
    at onaborted (/opt/app/backend/node_modules/express/lib/response.js:1052:15)
    at onfinish (/opt/app/backend/node_modules/express/lib/response.js:1088:50)
    at AsyncResource.runInAsyncScope (node:async_hooks:206:9)
    at listener (/opt/app/backend/node_modules/on-finished/index.js:170:15)
    at onFinish (/opt/app/backend/node_modules/on-finished/index.js:101:5)
    at callback (/opt/app/backend/node_modules/ee-first/index.js:55:10)
    at Socket.onevent (/opt/app/backend/node_modules/ee-first/index.js:93:5)
    at Socket.emit (node:events:530:35)
    at emitErrorNT (node:internal/streams/destroy:169:8)
    at emitErrorCloseNT (node:internal/streams/destroy:128:3) {
  code: 'ECONNABORTED'
}

**Comment by axiomen** [comment]
Do you still have `nginx-unathorized` jail disabled, or have you re-enabled it?

**Comment by babsz02** [comment]
yes the jail nginx-unauthorized is disabled

**Comment by axiomen** [comment]
That is strange. This is how I have it set up currently:

**Image**: `stonith404/pingvin-share:development`

`./swag/fail2ban/jail.local`:

```
[...]

[nginx-unauthorized]
enabled = false
port = http,https
filter = nginx-unauthorized
logpath = /config/log/nginx/access.log

[...]
```

`./swag/nginx/proxy-confs/pingvin-share.subdomain.conf`:

```
## Version 2023/05/31
# make sure that your pingvin-share container is named pingvin-share
# make sure that your dns has a cname set for pingvin-share

server {
    listen 443 ssl http2;
    listen [::]:443 ssl http2;

    server_name REDACTED.*;

    include /config/nginx/ssl.conf;

    client_max_body_size 0;

    location / {
        include /config/nginx/proxy.conf;
        include /config/nginx/resolver.conf;
        set $upstream_app REDACTED;
        set $upstream_port 3000;
        set $upstream_proto http;
        proxy_pass $upstream_proto://$upstream_app:$upstream_port;

    }

}
```


For this particular error, please refer to the #354, since the `401 Unathorized` error is a separate issue.

**Comment by babsz02** [comment]
We have a simular configuration
attention your jail is deactivated ( false )

Did you do my modification of the nginx-unauthorized filter parameter ?

**Comment by stonith404** [comment]
The should be fixed and will be released in the next release ASAP.

**Comment by stonith404** [comment]
Fixed in `v0.22.2`.
